records_mover/db/postgres/postgres_copy_options.py

```diff
@@ -0,0 +1,575 @@
+#  separately or together. The keywords Euro and European are
+#  synonyms for DMY; the keywords US, NonEuro, and NonEuropean are
+#  synonyms for MDY. See Section 8.5 for more information. The
+#  built-in default is ISO, MDY, but initdb will initialize the
+#  configuration file with a setting that corresponds to the
+#  behavior of the chosen lc_time locale.
+DateInputStyle = Union[Literal["DMY"], Literal["MDY"]]
+
+# https://www.postgresql.org/docs/9.2/sql-copy.html
+
+# https://www.postgresql.org/docs/9.3/multibyte.html
+postgres_encoding_names = {
+    'UTF8': 'UTF8',
+    # UTF-16 not supported
+    # https://www.postgresql.org/message-id/20051028061108.GB30916%40london087.server4you.de
+    'LATIN1': 'LATIN1',
+    'CP1252': 'WIN1252',
+}
+
+
+def needs_csv_format(hints: RecordsHints) -> bool:
+    # This format option is used for importing and exporting the Comma
+    # Separated Value (CSV) file format used by many other programs,
+    # such as spreadsheets. Instead of the escaping rules used by
+    # PostgreSQL's standard text format, it produces and recognizes
+    # the common CSV escaping mechanism.
+
+    # The values in each record are separated by the DELIMITER
+    # character. If the value contains the delimiter character, the
+    # QUOTE character, the NULL string, a carriage return, or line
+    # feed character, then the whole value is prefixed and suffixed by
+    # the QUOTE character, and any occurrence within the value of a
+    # QUOTE character or the ESCAPE character is preceded by the
+    # escape character. You can also use FORCE_QUOTE to force quotes
+    # when outputting non-NULL values in specific columns.
+    if hints['header-row'] or (hints['quoting'] is not None):
+        return True
+
+    return False
+
+
+def postgres_copy_options_common(unhandled_hints: Set[str],
+                                 hints: RecordsHints,
+                                 fail_if_cant_handle_hint: bool,
+                                 postgres_options: PostgresCopyOptions) -> Optional[DateInputStyle]:
+    date_input_style: Optional[DateInputStyle] = None
+
+    # https://www.postgresql.org/docs/9.5/datatype-datetime.html#DATATYPE-DATETIME-INPUT
+
+    # "Date and time input is accepted in almost any reasonable
+    # format, including ISO 8601, SQL-compatible, traditional
+    # POSTGRES, and others. For some formats, ordering of day, month,
+    # and year in date input is ambiguous and there is support for
+    # specifying the expected ordering of these fields. Set the
+    # DateStyle parameter to MDY to select month-day-year
+    # interpretation, DMY to select day-month-year interpretation, or
+    # YMD to select year-month-day interpretation."
+
+
+    # $ ./itest shell
+    # $ db dockerized-postgres
+    # postgres=# SHOW DateStyle;
+    #  DateStyle
+    # -----------
+    #  ISO, MDY
+    # (1 row)
+    #
+    # postgres=#
+
+    # TODO: File GitHub issue documenting that we're assuming 'ISO,
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker